vault/router.go

```diff
@@ -91,6 +91,15 @@ func (r *Router) Remount(src, dst string) error {
 
 	// Update the mount point
 	r.root.Delete(src)
+	mountEntry, ok := raw.(*mountEntry)
```

Comment:
No need to guard the cast here since we control the inputs. We should maybe make a private `remount` method on `dynamicSystemView` and just invoke that to encapsulate the logic.